homeassistant/components/ihc/ihcdevice.py

```diff
@@ -41,21 +65,52 @@ def name(self):
         """Return the device name."""
         return self._name
 
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return f"ihc{self.controller_id}{self.ihc_id}"
```

Comment:
We don't need to add the integration domain to the unique_id. The entity registry is aware of both the integration and the platform domain.

Reply:
I was not aware of this - I would prefer to keep as this because a lot of users have already been using this in a custom integration for testing, so if I change it all the unique ids will change for them.

Reply:
No. If you want to support a custom integration, which I don't recommend, you should update the custom integration and migrate the unique_id to our standard.

Here's a helper that works for config entry integrations:
https://github.com/home-assistant/core/blob/a0895f1bf23f72e7689328165e68f4e3450c91ab/homeassistant/helpers/entity_registry.py#L889-L893

Reply:
I have removed "ihc" from the unique id